ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -84,13 +84,14 @@ object Types {
   case object NullType extends Type
 
   /** Reference types (allowed for classOf[], is/asInstanceOf[]). */
-  sealed abstract class ReferenceType extends Type
+  sealed trait ReferenceType
```

Comment:
Any reason you do not add a `toType` method here that essentially returns `this` (with `this` having a self-type of `Type with ReferenceType`)? It seems its less fragile than the casts but essentially does the same.